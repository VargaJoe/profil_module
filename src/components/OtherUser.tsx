
import * as React				            from 'react';
import { Link } 					        from 'react-router-dom';
import { connect }                          from 'react-redux';
import { Actions, 
    // Reducers 
}                                           from '@sensenet/redux';
// import { PathHelper }                       from '@sensenet/client-utils';

// save config 
const DATA = require('../config.json');

interface Props {
    getUsers: Function;
}

interface Stats {
    users: any;
    isDataFetched: boolean;    
}

class OtherUser extends React.Component<Props, Stats> {

    constructor(props: Props) {
        super(props);
        this.state = {
            users: null,
            isDataFetched: false
        };
    }

    componentDidMount  () {
        let path = DATA.ims;
        const  options = {
            select : ['Name', 'DisplayName', 'JobTitle', 'Email', 'Skype'],
        };
        let users = this.props.getUsers(path, options);

        users.then( (result: any) => {
            console.log(result);
            this.setState({
                isDataFetched : true,
                users: result.value.entities.entities
            });
        });

        users.catch((err: any) => {
            console.log(err);
        });
    }

    render() {
        if ( !this.state.isDataFetched ) {
            return null;
        }
        let usersList = this.state.users;
        const AllUsers = Object.keys(usersList).map( (key: any) => 
            (
                <div key={key}>
                    Name: 
                    <Link to={'/user/:' + usersList[key].Name} >
                        {usersList[key].Name}
                    </Link>
                    <br/>
                    Email: {usersList[key].Email}
                    <br/>
                    Position: {usersList[key].JobTitle}
                    <br/>
                    <br/>
                </div>
            )
        );
        
		return (
            
			<div>
				test OtherUser with routing
                <br/>
                <br/>
                {AllUsers}
                <Link to="/" >
                    Back Home
                </Link>
			</div>
		);
	}
} 

const mapStateToProps = (state: any, match: any) => {
    return {
        userName :      state.sensenet.session.user.userName,
    };
};

export default connect(
    mapStateToProps,
    (dispatch) => ({
        getUsers:    (path: string, options: any) => dispatch(Actions.requestContent( path, options )),
    })
)(OtherUser as any);
 