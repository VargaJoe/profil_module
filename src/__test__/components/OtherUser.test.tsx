jest.unmock('../../index.tsx');
jest.unmock('redux-mock-store');

import * as React 							from 'react';
import OtherUser 							from '../../components/OtherUser';
import { 
	configure, 
	shallow,
	mount, 
	render,
	ShallowWrapper } 						from 'enzyme';
import * as Adapter 						from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';
import * as createMockStore 				from 'redux-mock-store';
import { Actions }                          from '@sensenet/redux';

const initialState = {};
const DATA = require('../../config.json');

configure( {adapter: new Adapter()} );

describe('<Menu /> shallow rendering', () => {
	let store, otheruser: ShallowWrapper<any, any>, otheruserRender: Cheerio;
	let path = DATA.ims;
	const  options = {
		select : ['Name', 'DisplayName', 'JobTitle', 'Email', 'Skype'],
		query: 'TypeIs:User',
	};
	const mockStore = (createMockStore as any)();
	beforeEach( () => {
        store = mockStore({
			sensenet: {
				session: {
					user: {
						userName: 'TestUser1'
					}
				}
			}
		});
		otheruser = shallow(<OtherUser store={store}/> );  
		otheruserRender = render(<OtherUser store={store}/> ); 
    }); 
	
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(otheruserRender).toMatchSnapshot();
	});

	// test Snapshot 
	it('Match to snapshot', () => {
		otheruser.setState({
			users: [{
				Name: 'some Name',
				Email: 'email',
				JobTitle: 'Mothing'
			},
			{
				Name: 'some Name2',
				Email: 'email2',
				JobTitle: 'Mothing2'
			}]
		});
		expect(otheruserRender).toMatchSnapshot();
	});

	it('Props name value should be the same as in store', () => {
        expect(otheruser.props().userName).toBe('TestUser1');
	});
	 
	it('Get all users action type', () => {
		const getUsers = otheruser.props().getUsers(path, options);
		expect(getUsers.type).toEqual('FETCH_CONTENT');
	});

	test(`on a React component that loads data into state in componentDidMount`, async () => {
		const wrapper: ShallowWrapper<any, any> = shallow(<OtherUser  store={store}/>);

		const spy = jest.spyOn(wrapper.instance(), 'componentDidMount');
		const ddd = wrapper.instance().componentDidMount();
		expect(spy).toHaveBeenCalledTimes(1);
	});
}); 
 