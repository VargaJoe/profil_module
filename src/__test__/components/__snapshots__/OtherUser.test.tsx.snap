// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> shallow rendering Match to snapshot 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Menu /> shallow rendering Match to snapshot 2`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
